### DISCUSSION TITLE
Quarkus Reactive Messaging RabbitMQ in status "Preview"

### DISCUSSION THREAD

**Discussion by soframel** [quarkusio/quarkus#39632] [Q&A] [answered]
Hello,

I have a question concerning the status of the Reactive Messaging RabbitMQ extension, because it is set to be in "Preview", which raises some questions in my company: 
Do you think there could be features that could be changed in the near future because of this "Preview" status? Because it seems to rely heavily on Smallrye Reactive Messaging Rabbitmq, which is apparently stable (as there are no warnings on the site and it is a version 3.*). The custom features seem to be related to dev-ui only, and perhaps just the passing of credentials? Do you think there is a high risk of incompatible changes (outside dev-ui, let's say) in the future?

In addition, do you have an idea of when this status would be changed to "Stable"?

Thank you

[a comment by an automated account is not shown]

**Reply by cescoffier** [reply, marked as answer]
It's in preview (as well as the reactive messaging connector BTW) because we do not have a deep expertise on rabbitmq. We are looking for contributors which could improve how we handle back pressure, improve the dev service... 

The main features are not going to change, but I'm sure there are things we can do better.

**Reply by impproductions** [reply]
Sorry for reviving this thread, but we're about to introduce Quarkus in our stack and we need to understand if it is a good fit in our Rabbit-based environment.

Regardless of its stability, is the extension production ready? If it's not, what's the recommended way to work with rabbit in Quarkus for production workloads?

We're using rabbit 3.8.35 (AMQP 0-9-1)

Thank you for your help

**Reply by cescoffier** [reply]
@ozangunalp did a lot of improvements recently. So, I would say yes. However, we are still not RabbitMQ experts.
